synapse/handlers/federation_event.py

```diff
@@ -1010,9 +1010,8 @@ async def _handle_marker_event(self, origin: str, marker_event: EventBase) -> No
         room_version = await self._store.get_room_version(marker_event.room_id)
         create_event = await self._store.get_create_event_for_room(marker_event.room_id)
         room_creator = create_event.content.get(EventContentFields.ROOM_CREATOR)
-        if (
-            not room_version.msc2716_historical
-            or not self._config.experimental.msc2716_enabled
+        if not room_version.msc2716_historical and (
+            not self._config.experimental.msc2716_enabled
             or marker_event.sender != room_creator
         ):
```

Comment:
We only want to process MSC2716 events when:

 - The room version supports MSC2716
 - Any room version with the `msc2716_enabled` experimental feature enabled and the event is from the room creator

Reply:
I'm a bit confused (and haven't been following this work too closely -- so sorry if this is a dumb question), but what's the relationship between the room version supporting MSC2716 vs. having MSC2716 enabled in the config? I would think the latter implied the former, but it sounds like they're completely different?

I don't see how a room version can have `msc2716_enabled` since it is a configuration flag?

Reply:
@clokep Since we're considering MSC2716 pretty backwards compatible, we're allowing MSC2716 things to happen in current room versions but the homeserver still needs the `experimental.msc2716_enabled` config enabled and be the room creator to avoid power level intricacies.

In room versions that support MSC2716, we allow everything regardless.

Reply:
Should the first comment read:

> We only want to process MSC2716 events when:
> 
> * The room version supports MSC2716
> * Any room ~~version~~ with the `msc2716_enabled` experimental feature enabled and the event is from the room creator

Reply:
@clokep Either seems valid but I could see how that would be more clear 👍. I was trying to convey that it works in all of the other current room versions.

Reply:
Alright. Thanks.